Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -2553,13 +2672,22 @@ void RenderBlockFlow::computeLogicalLocationForFloat(FloatingObject& floatingObj
                 logicalLeftOffset = logicalLeftOffsetForContent(logicalTopOffset); // Constant part of left offset.
                 floatLogicalWidth = std::min(logicalWidthForFloat(floatingObject), logicalRightOffset - logicalLeftOffset);
             }
+            availableSpace = logicalRightOffsetForPositioningFloat(logicalTopOffset, logicalRightOffset, false, &heightRemainingRight) - floatLogicalLeft;
         }
         floatLogicalLeft = std::max(logicalLeftOffset - borderAndPaddingLogicalLeft(), floatLogicalLeft);
+        
+        if (containingBlockMarginTrim.contains(MarginTrimType::InlineStart) && logicalLeftOffset == floatLogicalLeft)
+            trimMarginForFloat(floatingObject, MarginTrimType::InlineStart);
+        if (containingBlockMarginTrim.contains(MarginTrimType::InlineEnd) && floatLogicalLeft + logicalWidthForFloat(floatingObject) == logicalRightOffset)
+            trimMarginForFloat(floatingObject, MarginTrimType::InlineEnd);
```

Comment:
> Discarding the inline-start/inline-end margin of an inline-start/inline-end float (or equivalent) whose outer edge on that side coincides with the inner edge of the container.
I am not super good at interpreting spec in general (so please bear with me), but I am wondering if trimming an inline-end of an inline-start float is really what we are supposed to do.

Reply:
Yeah so my reasoning behind this because one of the introductory sentences for the property was: `This property allows the container to trim the margins of its children where they adjoin the container’s edges.`

However, I suppose since the authors specified the behavior this way for floats we should follow it that was

So in this case:

```
<div>
<div style="float: left; width: 50px; height: 50px; margin-inline: 50px; background-color: green;">
</div>
</div>
```
only the inline-start margin would be trimmed and the remaining inline-end margin would remain. Similar changes would need to be made for the inline layout portion as well

Reply:
Ended up removing the logic so now it follows exactly what the spec says.